api/src/http/server/middleware.rs

```diff
@@ -0,0 +1,70 @@
+use axum::{extract::Request, middleware::Next, response::Response};
+use axum_extra::extract::CookieJar;
+use chrono::Utc;
+use jsonwebtoken::{decode, DecodingKey, Validation};
+use serde::{Deserialize, Serialize};
+use uuid::Uuid;
+
+use crate::http::server::ApiError;
+
+#[derive(Clone, Debug)]
+pub struct AuthState {
+    pub user_id: Uuid
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Claims {
+    pub sub: Uuid, // user_id
+    pub exp: i64, // expiration timestamp
+    pub iat: i64, // issued at timestamp
+}
+
+impl Claims {
+    pub fn is_expired(&self) -> bool {
+        self.exp < Utc::now().timestamp()
+    }
+}
+
+pub async fn auth_middleware(
+    cookie_jar: CookieJar,
+    mut request: Request,
+    next: Next,
+) -> Result<Response, ApiError> {
+    // try to get token from cookies
+    let auth_cookie = cookie_jar
+        .get("access_token");
+
+    let token: String;
+
+    if auth_cookie.is_none() {
+        return Err(ApiError::AuthenticationError("Missing JWT token".to_string()));
+    } else {
+        // extract token from cookie
+        token = auth_cookie.unwrap().value().to_string();
+    }
```

Comment:
```suggestion
    auth_cookie.ok_or_else(|| ApiError::AuthenticationError("Missing JWT token".to_string()))?;
```

Could you use that for the handling of none value.